### PR TITLE
:sparkles: Enable transit encryption to S3 bucket

### PR DESCRIPTION
**Pull request by fiunchinho**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [X] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Enable transit encryption on S3 buckets
```

### CODE REVIEW

**Comment by faiq** [comment]
Does this require any changes for IAM permissions on the roles

**Comment by fiunchinho** [comment]
> Does this require any changes for IAM permissions on the roles

[Official docs](https://aws.amazon.com/blogs/storage/enforcing-encryption-in-transit-with-tls1-2-or-higher-with-amazon-s3/) don't mention anything about IAM so I'd say no, it doesn't require any changes on IAM permissions.

**Review by AndiDog — COMMENTED** [review]
/lgtm

File: pkg/cloud/services/s3/s3.go [hunk 1]
```diff
@@ -306,6 +306,20 @@ func (s *Service) bucketPolicy(bucketName string) (string, error) {
...
+			Sid:    "ForceSSLOnlyAccess",
+			Effect: iam.EffectDeny,
+			Principal: map[iam.PrincipalType]iam.PrincipalID{
+				iam.PrincipalAWS: []string{"*"},
+			},
+			Action:   []string{"s3:*"},
+			Resource: []string{fmt.Sprintf("arn:%s:s3:::%s/*", partition, bucketName)},
+			Condition: iam.Conditions{
+				"Bool": map[string]interface{}{
+					"aws:SecureTransport": false,
```
Comment (richardcase) [inline pkg/cloud/services/s3/s3.go:319]: Should this be set to **true**?

Comment (fiunchinho) [reply pkg/cloud/services/s3/s3.go:319]: The `Effect` of the policy is to `deny` requests when the condition `SecureTransport: false` is met. Hence only allowing requests with TSL. Does that make sense?

Comment (richardcase) [reply pkg/cloud/services/s3/s3.go:319]: Ah, yes. That makes sense....thanks @fiunchinho

**Pull request merged**